awx/main/tasks/receptor.py

```diff
@@ -338,6 +338,11 @@ def _run_internal(self, receptor_ctl):
         if os.path.exists(artifact_dir):
             shutil.rmtree(artifact_dir)
 
+        # If we happen to leave around any pipes that could cause hangs when unstreaming
+        key_data_file = os.path.join(self.runner_params['private_data_dir'], 'artifacts', str(self.task.instance.id), 'ssh_key_data')
+        if os.path.exists(key_data_file):
+            os.remove(key_data_file)
```

Comment:
Would it make more sense to push this path onto `self.task.cleanup_paths` instead?

Reply:
the cleanup paths will only cleanup after the job reaches the finished status and the `run` method exits. For this bug, we need to remove the pipe before the _artifacts_ are unstreamed, which happens inside of the `run` method, so before the files are cleaned up.

So it's all about timing of _when_ they get cleaned up, however, this PR didn't fix the issue. @fosterseth has a patch that cleans up at a slightly different time which seems to be working.